.github/workflows/catalog-updates-check.yaml

```diff
@@ -0,0 +1,67 @@
+name: Check for unsafe catalog updates
+"on":
+  pull_request:
+    types: [opened, synchronize, reopened, edited]
+jobs:
+  check_catalog_correctly_updated:
+    name: Check updates to latest-dev and reverse-dev are properly handled by PR
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0
+      - name: Install pglast
+        run: |
+          python -m pip install pglast
+      - name: Check latest-dev contents
+        shell: bash {0}
+        id: check_latestdev
+        env:
+          BODY: ${{ github.event.pull_request.body }}
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          echo "$BODY" | egrep -qsi '^disable-check:.*\<catalog-updates\>'
+
+          if [[ $? -ne 0 ]]; then
+            if ! python scripts/check_using_ast.py "sql/updates/latest-dev.sql"; then
+              echo
+              echo "To disable the catalog updates check, add this trailer to pull request message:"
+              echo
+              echo "Disable-check: catalog-updates"
+              echo
+              echo "Trailers follow RFC2822 conventions, so no whitespace"
+              echo "before field name and the check is case-insensitive for"
+              echo "both the field name and the field body."
+              echo "printed_message=1" >> $GITHUB_OUTPUT
+              exit 1
+            fi
+          fi
+          echo "printed_message=0" >> $GITHUB_OUTPUT
+          exit 0
+      - name: Check reverse-dev contents
+        if: always()
+        shell: bash {0}
+        env:
+          BODY: ${{ github.event.pull_request.body }}
+        run: |
+          echo "$BODY" | egrep -qsi '^disable-check:.*\<catalog-updates\>'
+          if [[ $? -ne 0 ]]; then
+            if ! python scripts/check_using_ast.py "sql/updates/reverse-dev.sql"; then
+              if [[ ${{ steps.check_latestdev.outputs.printed_message }} == 0 ]]; then
+                echo
+                echo "To disable the catalog updates check, add this trailer to pull request message:"
+                echo
+                echo "Disable-check: catalog-updates"
+                echo
+                echo "Trailers follow RFC2822 conventions, so no whitespace"
+                echo "before field name and the check is case-insensitive for"
+                echo "both the field name and the field body."
+                echo "printed_message=1" >> $GITHUB_OUTPUT
+                exit 1
```

Comment:
Why do we need this? Shouldnt those constraints always be enforced?

Reply:
Can you check if @extschema@ is working correctly. You might have to add a preprocessing step if the sql parser doesnt accept it.

Reply:
yes, this should be a required check, and I don’t think we would get any false positives here, so I removed this. If we realize there are false positives or cases where we’d need to skip the check, I can re-add it.